src/services/theme.hooks.ts

```diff
@@ -6,10 +6,13 @@
 import { useState, useContext } from 'react';
 import { ThemeContext } from '../frontend/app/app-context-theme/app-context-theme';
 import themes from '../shared/themes';
+import { useAuth } from './auth.hooks';
 
 // Provider hook that creates theme object and handles state
 export const useProvideTheme = () => {
-  const [theme, setTheme] = useState(themes[0]);
+  const auth = useAuth();
+  const defaultThemeIndex = themes.findIndex((t) => auth.user?.defaultTheme === t.name);
+  const [theme, setTheme] = useState(themes[defaultThemeIndex > -1 ? defaultThemeIndex : 0]);
```

Comment:
Sorta still confused how this helps / what this intended to do? Is this intended to solve the dev user default theme not working?

The way I set it up was
1. visit login page
2. default dark always used via index 0
3. auth success pulls default theme and toggles as-needed
4. auth'd app loads with user's default theme`theme`auth success pulls default theme and toggles as-needed
5. auth'd app loads with user's default theme`default dark used
6. auth success pulls default theme and toggles as-needed
7. auth'd app loads with user's default theme

does this addition suggest that the theme context is rebuilt whenever auth changes? wouldn't that be redundant with toggling theme in login success call? or perhaps at least the remove the login success theme toggle in favor of this?